src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -113,9 +113,11 @@ private IEnumerable<string> ExpandGlobbedUrl(string include, string exclude)
                 if (!Cache.TryGetValue(cacheKey, out files))
                 {
                     var options = new MemoryCacheEntryOptions();
-                    foreach (var pattern in includePatterns)
+
+                    // Perf: Avoid allocating enumerator
+                    for (var i = 0; i < includePatterns.Length; i++)
                     {
-                        var changeToken = FileProvider.Watch(pattern);
+                        var changeToken = FileProvider.Watch(includePatterns[i]);
                         options.AddExpirationToken(changeToken);
```

Comment:
I'd remove the comment if it's not really perf-sensitive code. The point about for/foreach I don't really mind either way. I don't think a `for` makes the code worse.